src/tools/rbd_mirror/image_replayer/BootstrapRequest.h

```diff
@@ -52,14 +52,15 @@ class BootstrapRequest : public BaseRequest {
         Journaler *journaler,
         MirrorPeerClientMeta *client_meta,
         Context *on_finish,
-        ProgressContext *progress_ctx = nullptr) {
+        ProgressContext *progress_ctx = nullptr,
+        bool *do_resync = nullptr) {
```

Comment:
I was not sure about my approach so wanted it to be as less intrusive as possible. Now seeing your comment that it is ok, I will do make do_resync non-optional.